### PR TITLE
Refactor include directories and doc repo via README files

### PR DESCRIPTION
**Pull request by netoptimizer**
This is my Work-in-Progress refactoring of the include directories and documenting the repo via README files.
I will push more to this branch.

### CODE REVIEW

**Comment by netoptimizer** [comment]
I'm going to merge this soon, unless someone manage to object.
There are too many adjustments and I don't that to get too much out of sync.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Practical BPF examples
+
+This git repository contains a diverse set of **practical BPF examples** that
+solve (or demonstrate) a specific use-case using BPF.
+
+It is meant to ease doing **rapid prototyping and development**, writing C-code
+BPF programs using libbpf.  The goal is to make is **easier for developers** to
+get started coding.
+
+Most developers struggle to get a working BPF build environment.  The repo
```
Comment (tohojo) [inline README.md:10]: s/most/many/ ? :)

**Review by tohojo — APPROVED** [review]
One small nit in the text, otherwise LGTM :)

File: README.md (hunk 1 above)
Comment (netoptimizer) [reply README.md:10]: s/Most/All/  ? :-P

**Comment by netoptimizer** [comment]
Toke approved this and we need to get this merged before @freysteinn fork the repo.

**Pull request merged**